From 03ba4ac32a737d17a595b95bb2abc3f2650e49ec Mon Sep 17 00:00:00 2001
From: Mike Snitzer <snitzer@redhat.com>
Date: Fri, 28 Nov 2014 16:18:31 -0500
Subject: [md] dm-array: if resizing the array is a noop set the new root to the old one

Message-id: <1417191518-16689-3-git-send-email-snitzer@redhat.com>
Patchwork-id: 101004
O-Subject: [RHEL7.1 PATCH 2/9] dm array: if resizing the array is a noop set the new root to the old one
Bugzilla: 1165050
RH-Acked-by: Joe Thornber <thornber@redhat.com>
RH-Acked-by: Heinz Mauelshagen <heinzm@redhat.com>

BZ: 1165050

Upstream linux-dm.git commit 64441799dc74a24e54ca0973a29c3181a6284130
Author: Joe Thornber <ejt@redhat.com>
Date:   Mon Nov 24 14:08:57 2014 +0000

    dm array: if resizing the array is a noop set the new root to the old one

    This could've been quite bad (to return success but not update the new
    root to point at the old) but in practice the only known consumer of the
    dm array code is the DM cache target.  And the DM cache target passes in
    the same old root to array_resize() anyway.

    Signed-off-by: Joe Thornber <ejt@redhat.com>
    Signed-off-by: Mike Snitzer <snitzer@redhat.com>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/md/persistent-data/dm-array.c b/drivers/md/persistent-data/dm-array.c
index 1d75b1d..e64b61a 100644
--- a/drivers/md/persistent-data/dm-array.c
+++ b/drivers/md/persistent-data/dm-array.c
@@ -645,8 +645,10 @@ static int array_resize(struct dm_array_info *info, dm_block_t root,
  int r;
  struct resize resize;
 
- if (old_size == new_size)
+ if (old_size == new_size) {
+  *new_root = root;
   return 0;
+ }
 
  resize.info = info;
  resize.root = root;
-- 
1.7.1